microdata.py

```diff
@@ -104,8 +113,11 @@ def json_dict(self):
 
 class URI(object):
 
-    def __init__(self, string):
-        self.string = string
+    def __init__(self, string, domain=""):
+        if string.startswith("http://") or string.startswith("https://"):
```

Comment:
This feels like it's duplicating the stdlib `urljoin` semantics:

```python
>>> from urlparse import urljoin
>>> urljoin('https://example.com/foo/bar', '/baaz')
'https://example.com/baaz'
>>> urljoin('https://example.com/foo/bar', 'http://quux/baaz')
'http://quux/baaz'
```

Reply:
A few questions:

1. Judging by the `.travis.yml` this project is officially on Python 3. As such I should use `urllib.parse` right?
2. Sorry I'm not sure I agree with you that this bit is duplicating `urljoin` semantics. This conditional just ensures that the string starts with either "http" or "https". `urljoin` does not do that:

```
Python 3.5.2 (default, Nov 12 2018, 13:43:14) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from urllib.parse import urljoin
>>> urljoin("github.com", "edsu")
'edsu'
```

Can you clarify if I'm missing something here?

Reply:
[urljoin expects its first parameter to be a URL](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin):

```
>>> from urllib.parse import urljoin
>>> urljoin("https://github.com", "edsu")
'https://github.com/edsu'
```